SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -141,7 +141,7 @@ - (void)start {
         Class UIApplicationClass = NSClassFromString(@"UIApplication");
         BOOL hasApplication = UIApplicationClass && [UIApplicationClass respondsToSelector:@selector(sharedApplication)];
         if (hasApplication && [self shouldContinueWhenAppEntersBackground]) {
-            __weak __typeof__ (self) wself = self;
+            __weak typeof(self) wself = self;
```

Comment:
Also replace with `weakify`.

Reply:
The `@weakify(self)` will generate a temp variable like `self_weak_`. So, it's looks strange to change code like this.

```objectivec
[self_weak_ cancel];
```

If the caller does not use any weak-strong dance (only weak reference), using this `__weak` is OK.

Reply:
😅 Seems just a small typo. I just fix the `__typeof__` because it's not a standard C syntax (It's the GCC extension). Alough current clang also support `__typeof__`, it's better to use `typeof` instead.

Reply:
> The `@weakify(self)` will generate a temp variable like `self_weak_`. So, it's looks strange to change the original code like this after using `weakify`
> 
> ```objc
> [self_weak_ cancel];
> ```
> 
> or you can place a strongify it before usage.
> 
> ```objc
> @strongify(self)
> [self cancel];
> ```
> 
> So, if the original code, does not use any weak-strong dance (only weak reference), using this `__weak` is OK. No need to strongify it.

This weakSelf code is the last piece  in SDWebImage project after pull request #2651. How about add IDEWorkspaceChecks.plist to gitignore.